mcs/class/corlib/corert/Debug.cs

```diff
@@ -1,17 +1,22 @@
-namespace System.Diagnostics.Private
+
+namespace System.Diagnostics
 {
-	static class Debug
+	static partial class Debug
 	{
-		public static void Assert (bool condition)
-		{
-		}
+		internal static IDebugLogger s_logger = new MonoDebugLogger();
 
-		public static void Assert (bool condition, string message)
+		internal sealed class MonoDebugLogger : IDebugLogger
 		{
-		}
+			public void ShowAssertDialog(string stackTrace, string message, string detailMessage)
+			{
+				// FIXME should we g_error in that case?
+			}
 
-		public static void Fail (string message)
-		{
+			public void WriteCore(string message)
+			{
+				// FIXME replace by a call into the runtime, to g_critical for example
+				Console.Error.Write (message);
```

Comment:
I'm not sure we want stderr or g_critical here, wouldn't g_debug be the more equivalent?